### PR TITLE
[Svn] Implement support to install Command Line Tools on demand.

### PR DESCRIPTION
**Pull request by Therzok**
Need :eyes: from @slluis. When this is confirmed to be okay, I'll backport to C7.

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs [hunk 1]
```diff
@@ -198,11 +203,48 @@ public override SubversionBackend CreateBackend ()
...
+				if (Directory.Exists (path.Combine (".svn")))
+					return true;
+				path = path.ParentDirectory;
+			}
+			return false;
+		}
+
+		const string commandLineToolsSvn = "/Library/Developer/CommandLineTools/usr/lib/libsvn_client-1.0.dylib";
+		static bool IsDependentOnXcodeCLITools {
+			get { return Platform.IsMac && Environment.Is64BitOperatingSystem && !File.Exists (commandLineToolsSvn); }
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs]: Maybe make the file.exists evaluation lazy.

Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs]: And cache the result.

The command line tools installer can't be waited via WaitForExit, so we can't wait for the installer to finish then probe.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs [hunk 2]
```diff
@@ -198,11 +203,49 @@ public override SubversionBackend CreateBackend ()
...
+
+				var button = new AlertButton (GettextCatalog.GetString ("Install Command Line Tools"));
+				if (MessageService.AskQuestion (
+					GettextCatalog.GetString ("Xcode Command Line Tools are required to use Subversion."),
+					BrandingService.BrandApplicationName (
+						GettextCatalog.GetString ("Not installing will cause Subversion support to be disabled.\nPlease restart MonoDevelop afterwards to enable Subversion.")),
+					button,
+					AlertButton.Ok) == button) {
+					var p = Process.Start ("xcode-select", "--install");
+					p.WaitForExit ();
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs]: This starts another process, that can't be waited with `Process.WaitForExit()`, so we're asking the user to manually restart XS after the installation process.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix.csproj [hunk 3]
```diff
@@ -97,6 +97,10 @@
       <Name>Mono.Addins</Name>
       <Private>False</Private>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj">
+      <Project>{27096E7F-C91C-4AC6-B289-6897A701DF21}</Project>
+      <Name>MonoDevelop.Ide</Name>
+    </ProjectReference>
```
Comment (slluis) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix.csproj]: No local copy please.

Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix.csproj]: Mishap! Good catch!

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs [hunk 4]
```diff
@@ -198,11 +203,49 @@ public override SubversionBackend CreateBackend ()
...
+			// For Mac 64bits, we need to have Xcode Command Line Tools installed.
+			if (IsDependentOnXcodeCLITools.Value) {
+				if (!FallbackProbeDirectoryDotSvn (path))
+					return string.Empty;
+
+				var button = new AlertButton (GettextCatalog.GetString ("Install Command Line Tools"));
+				if (MessageService.AskQuestion (
+					GettextCatalog.GetString ("Xcode Command Line Tools are required to use Subversion."),
+					BrandingService.BrandApplicationName (
+						GettextCatalog.GetString ("Not installing will cause Subversion support to be disabled.\nPlease restart MonoDevelop afterwards to enable Subversion.")),
```
Comment (slluis) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Subversion.Unix/MonoDevelop.VersionControl.Subversion.Unix/SvnClient.cs]: Please change the messages to this:

**Text:**
This solution may be using Subversion. Do you want to install Xcode Command Line Tools now?

**Secondary Test:**
Xcode Command Line Tools are not currently installed, and are required to use Subversion.
Please restart MonoDevelop after the installation to enable Subversion support.

**Buttons:**
[Cancle] [Install]

**Pull request merged**